lib/url.js

```diff
@@ -563,6 +563,8 @@ Url.prototype.format = function() {
 
   if (this.query !== null && typeof this.query === 'object')
     query = querystring.stringify(this.query);
+  else
```

Comment:
Please add a typeof check to ensure that a string (and not something else) is passed.